frame/support/src/lib.rs

```diff
@@ -361,6 +364,19 @@ macro_rules! parameter_types {
 				I::from($value)
 			}
 		}
+
+		impl $crate::scale_info::TypeInfo for $name {
+			type Identity = Self;
+
+			fn type_info() -> $crate::scale_info::Type<$crate::scale_info::form::MetaForm> {
+				$crate::scale_info::Type::builder()
+					.path($crate::scale_info::Path::new(stringify!($name), module_path!()))
+					.composite(
+					    scale_info::build::Fields::unnamed()
+							.field(|f| f.ty::<$type>().type_name(stringify!($type)))
```

Comment:
is it corret to say the type info are those ?
The actual type is just a struct with no field and no generic.
I think we should use the regular `derive(TypeInfo)` here no ?

Reply:
Looking again, we might be able to remove the `TypeInfo` requirement for those constants, now we have `skip_type_params`. Looks like the constants are mostly used as a type argument to e.g. [`BoundedVec`](https://github.com/paritytech/substrate/blob/f612dd9f6a67c9e6f0f7d62c51641ca8664109cb/frame/support/src/storage/bounded_vec.rs#L41) which ends up in `PhantomData`.

Reply:
Ok brilliant I've been able to remove these entirely (and the `Get<u32> + TypeInfo` bounds) with a couple of `skip_type_params` attributes.